expression/constant.go

```diff
@@ -176,8 +176,13 @@ func (c *Constant) VecEvalJSON(ctx sessionctx.Context, input *chunk.Chunk, resul
 }
 
 func (c *Constant) getLazyDatum() (dt types.Datum, isLazy bool, err error) {
-	if c.ParamMarker != nil {
-		dt = c.ParamMarker.GetUserVar()
+	if p := c.ParamMarker; p != nil {
+		if p.ctx.GetSessionVars().StmtCtx.InExplainStmt {
+			// Since `ParamMarker` is not nil only in prepare/execute context, the query must be `explain for connection` when coming here.
+			// The PreparedParams may have been reset already, to avoid panic, we just use the pre-evaluated datum for this constant.
+			return
```

Comment:
```suggestion
			return dt, false, nil
```

maybe explicitly return a false `isLazy` is more readable?